packages/ember-metal/lib/property_set.js

```diff
@@ -78,10 +78,39 @@ export function set(obj, keyName, value, tolerant) {
     currentValue = obj[keyName];
   }
 
+  // TODO turn if block into a sveltable deprecated block
   if (isDescriptor(currentValue)) {
-    /* computed property */
-    currentValue.set(obj, keyName, value);
-  } else if (
+    deprecate(
+      `[DEPRECATED] computed property '${keyName}' was not set on object '${toString(
+        obj
+      )}' via 'defineProperty'`,
+      false,
+      {
+        id: 'ember-meta.descriptor-on-object',
+        until: '3.5.0',
+        url:
+          'https://emberjs.com/deprecations/v3.x#toc_use-defineProperty-to-define-computed-properties',
+      }
+    );
+
+    Object.defineProperty(obj, keyName, {
+      configurable: true,
+      enumerable: currentValue.enumerable === false,
+      get(keyName) {
+        currentValue.get(this, keyName);
+      },
+    });
+
+    meta(obj).writeDescriptors(keyName, currentValue);
+
+    if (typeof currentValue.setup === 'function') {
+      currentValue.setup(obj, keyName);
+    }
+
+    return currentValue.get(obj, keyName);
```

Comment:
Don’t we want to call `currentValue.set(obj, keyName, value)`? Why would we call `.get`?